### PR TITLE
Catch CLI exception and return exit code 1

### PR DESCRIPTION
**Pull request by zazabe**

### CODE REVIEW

File: library/CM/Cli/CommandManager.php [hunk 1]
```diff
@@ -299,8 +299,14 @@ protected function _getProcessWorkload($transactionName, CM_Cli_Command $command
                 } else {
                     $this->_outputError($this->getHelp());
                 }
+                exit(1);
```
Comment (njam) [inline library/CM/Cli/CommandManager.php]: This function should not exit() I think, only `bin/cm` should.
Have a look at `CommandManager::run()` which returns the exit code to use.

Comment (zazabe) [inline library/CM/Cli/CommandManager.php]: not sure to get it, the cmd is running in a fork, i think the only way to change the exit code of the CLI is via `CM_Process_Result::$_returnCode`, which is set in `CM_Process::_wait`, based on the child exit code (`pcntl_wexitstatus`)...
or you mean invalid arguments should not use an exit code = 1 ?

**Comment by njam** [comment]
We anyway want to remove forking capability from CLI-Manager (see https://github.com/cargomedia/cm/issues/2524). It can be done today - no CLI command requires forking anymore. Let's do that, it will make this much simpler.
Do you want to do it here? Or should we open a separate ticket?

**Comment by zazabe** [comment]
hum... just spent some time making tests working with forked cmd... about removing the forking, i think it's better to create a new PR

**Comment by njam** [comment]
Closing in favor of https://github.com/cargomedia/cm/issues/2585

**Pull request closed without merging**